src/cmd/singularity/actions_test.go

```diff
@@ -174,6 +178,44 @@ func testSTDINPipe(t *testing.T) {
 	}
 }
 
+// testRunFromURI tests min fuctionality for singularity run/exec URI://
+func testRunFromURI(t *testing.T) {
+	tests := []struct {
+		name   string
+		image  string
+		action string
+		argv   []string
+		execOpts
```

Comment:
Probably should just be called opts since we are also using `run` below.

Reply:
Agree, will change

Reply:
`actionOpts`  maybe?